bdsg/include/bdsg/snarl_distance_index.hpp

```diff
@@ -805,14 +978,18 @@ class SnarlDistanceIndex : public SnarlDecomposition, public TriviallySerializab
     const static bool is_externally_end_end_connected(const size_t tag) {return tag & 256;}
 
 
-    ////////////////////////////////  SnarlTreeRecord class for interpreting the records in a distance index
-
-    //Define a struct for interpreting each type of snarl tree node record (For node, snarl, chain)
-    //These will be used with net_handle_t's, which store a pointer into snarl_tree_records 
-    //as an offset. The last 4 bits of the net handle will be the connectivity of the handle,
-    //as a connectivity_t
-    //TODO: This might be overkill but I want it to be easy to change what gets stored in the index
-    //
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////  SnarlTreeRecord class for interpreting the records in a distance index
+//
+/* Define a struct for interpreting each type of snarl tree node record (For node, snarl, chain)
+ *
+ * This is meant to be a layer in between snarl_tree_records and the public interface
+ * Each net_handle_t has a pointer to the start of a record in snarl_tree_records. 
+ * SnarlTreeRecord keeps the pointer and interprets the values stored in the record 
+ *
+ * SnarlTreeRecordConstructor does the same thing but for writing values to the index
+ *
+ */
     struct SnarlTreeRecord {
```

Comment:
These are cheap abstractions, but I don't think they're quite zero-cost, since you need to spend time arranging the pointers and offsets into the right memory layout somewhere to form the object to call the methods, and following `this` pointers and possibly calling non-inlined functions. Just keeping everything in arguments to static `inline` functions to do the accesses might be slightly faster.